File: CODE_FILES/TASK3/PYTHON/task3.py
import random
import time
import bisect

# Generate data
data = [random.randint(0, 1_000_000) for _ in range(1_000_000)]

start = time.time()

data.sort()
found = 0

for _ in range(1000):
    target = random.randint(0, 1_000_000)
    index = bisect.bisect_left(data, target)
    if index < len(data) and data[index] == target:
        found += 1

end = time.time()

print(f"Found {found} values.")
print(f"Time taken: {(end - start) * 1000:.2f} ms")

File: CODE_FILES/TASK3/GO/task3.go
package main

import (
	"fmt"
	"math/rand"
	"sort"
	"time"
)

func main() {
	rand.Seed(time.Now().UnixNano())
	data := make([]int, 1_000_000)
	for i := range data {
		data[i] = rand.Intn(1_000_001)
	}

	start := time.Now()

	sort.Ints(data)
	found := 0

	for i := 0; i < 1000; i++ {
		target := rand.Intn(1_000_001)
		index := sort.SearchInts(data, target)
		if index < len(data) && data[index] == target {
			found++
		}
	}

	elapsed := time.Since(start)
	fmt.Printf("Found %d values.\n", found)
	fmt.Printf("Time taken: %d ms\n", elapsed.Milliseconds())

    	// Adding time to see it in Process Explorer
    	time.Sleep(3 * time.Second)
}
